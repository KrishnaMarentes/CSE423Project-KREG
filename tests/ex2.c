#include <stdio.h>
#include <stdlib.h>

int main() {
	int j = 4;
	int *f = &j;
	int m1 = 4 & 7;
	printf("testing out this       %d\n", *f);
	return 0;
}
