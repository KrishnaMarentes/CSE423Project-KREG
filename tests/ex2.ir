function foo(int i)
{
int m;
int j;
int KREG.6;

KREG.6 = 6 + i;
m = KREG.6;
return m;
}

function main()
{
int i;
int b;
int KREG.7;
int KREG.8;
int KREG.9;
int KREG.14;
int KREG.15;
int KREG.16;

KREG.7 = 12 * 18;
i = KREG.7;
KREG.8 = 2 + 4;
b = KREG.8;
KREG.9 = foo();
i = KREG.9;
goto <KREG.3>;
<KREG.4>:
if i > 1 goto <KREG.10> else goto <KREG.11>;
<KREG.10>:
KREG.14 = i;
i = i + 1;
KREG.15 = KREG.14;
goto <KREG.13>;
<KREG.11>:
if b == 40 goto <KREG.12> else goto <KREG.13>;
<KREG.12>:
i = i - 1;
goto <KREG.13>;
<KREG.13>:
if b < 0 goto <KREG.17> else goto <KREG.18>;
<KREG.17>:
b = 4;
goto <KREG.18>;
<KREG.18>:
<KREG.3>:
if i < 3 goto <KREG.4>;
<KREG.5>:

return b;
}

