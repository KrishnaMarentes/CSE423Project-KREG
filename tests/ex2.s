.section .text

.global foo
foo:
	push %ebp
	mov %esp, %ebp
	sub $28, %esp
	movl $2, -4(%ebp)
	movl $1, -8(%ebp)
	movl $3, -12(%ebp)
	movl $3, -16(%ebp)
	mov $2, %ecx
	mov 8(%ebp), %edx
	imul %ecx, %edx
	mov %edx, %ecx
	mov $3, %edx
	add %ecx, %edx
	mov %edx, %ecx
	mov 12(%ebp), %edx
	sub %ecx, %edx
	mov %edx, %eax
	leave
	ret
.global main
main:
	push %ebp
	mov %esp, %ebp
	sub $8, %esp
	push $22
	push $5
	call foo
	mov %eax, -4(%ebp)
	leave
	ret
