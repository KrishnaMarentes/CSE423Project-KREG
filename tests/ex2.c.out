program 
  declarationList 
    declaration 
      funDeclaration 
        typeSpecifier int
        main(params)
        compoundStmt {
          statementList 
            statementList 
              statementList 
                statementList 
                  statementList 
                    statementList 
                      statementList statementList
                        varDeclaration 
                          typeSpecifier int
                          varDeclList 
                            varDeclList 
                              varDeclInitialize 
                                varDeclId j
                                =
                                expression 
                                  simpleExpression 
                                    andExpression 
                                      unaryRelExpression 
                                        relExpression 
                                          sumExpression 
                                            mulExpression 
                                              unaryExpression 
                                                factor 
                                                  immutable 
                                                    constant 4
                            ,
                            varDeclInitialize 
                              varDeclId m
                          ;
                      varDeclaration 
                        typeSpecifier int
                        varDeclList 
                          varDeclInitialize 
                            varDeclId f
                            =
                            expression 
                              simpleExpression 
                                andExpression 
                                  unaryRelExpression 
                                    relExpression 
                                      sumExpression 
                                        mulExpression 
                                          unaryExpression 
                                            factor 
                                              mutable j
                        ;
                    varDeclaration 
                      typeSpecifier int
                      varDeclList 
                        varDeclInitialize 
                          varDeclId m1
                          =
                          expression 
                            immutable 
                              constant 4
                            &
                            expression 
                              simpleExpression 
                                andExpression 
                                  unaryRelExpression 
                                    relExpression 
                                      sumExpression 
                                        mulExpression 
                                          unaryExpression 
                                            factor 
                                              immutable 
                                                constant 7
                      ;
                  varDeclaration 
                    typeSpecifier int
                    varDeclList 
                      varDeclInitialize 
                        varDeclId k2
                        =
                        expression 
                          immutable 
                            constant 4
                          ^
                          expression 
                            simpleExpression 
                              andExpression 
                                unaryRelExpression 
                                  relExpression 
                                    sumExpression 
                                      mulExpression 
                                        unaryExpression 
                                          factor 
                                            immutable 
                                              constant 7
                    ;
                statement 
                  expressionStmt 
                    expression 
                      mutable k2
                      ^=
                      expression 
                        simpleExpression 
                          andExpression 
                            unaryRelExpression 
                              relExpression 
                                sumExpression 
                                  mulExpression 
                                    unaryExpression 
                                      factor 
                                        immutable 
                                          constant 9
                    ;
              statement 
                expressionStmt 
                  expression 
                    simpleExpression 
                      andExpression 
                        unaryRelExpression 
                          relExpression 
                            sumExpression 
                              mulExpression 
                                unaryExpression 
                                  factor 
                                    immutable 
                                      call printf(
                                        args 
                                          argList 
                                            argList 
                                              expression 
                                                simpleExpression 
                                                  andExpression 
                                                    unaryRelExpression 
                                                      relExpression 
                                                        sumExpression 
                                                          mulExpression 
                                                            unaryExpression 
                                                              factor 
                                                                immutable 
                                                                  constant "testing out this       %d\n"
                                            ,
                                            expression 
                                              simpleExpression 
                                                andExpression 
                                                  unaryRelExpression 
                                                    relExpression 
                                                      sumExpression 
                                                        mulExpression 
                                                          unaryExpression 
                                                            unaryop *
                                                            unaryExpression 
                                                              factor 
                                                                mutable f
                                        )
                  ;
            statement 
              returnStmt return
                expression 
                  simpleExpression 
                    andExpression 
                      unaryRelExpression 
                        relExpression 
                          sumExpression 
                            mulExpression 
                              unaryExpression 
                                factor 
                                  immutable 
                                    constant 0
                ;
          }
  <EOF>

program 
  funDeclaration 
    typeSpecifier int
    main
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclList 
          varDeclInitialize 
            varDeclId j
            =
            constant 4
        ,
        varDeclInitialize 
          varDeclId m
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclInitialize 
          varDeclId f
          =
          mutable j
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclInitialize 
          varDeclId m1
          =
          expression 
            constant 4
            &
            constant 7
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclInitialize 
          varDeclId k2
          =
          expression 
            constant 4
            ^
            constant 7
    expression 
      mutable k2
      ^=
      constant 9
    call printf
      args 
        argList 
          argList 
            constant "testing out this       %d\n"
          ,
          unaryExpression 
            unaryop *
            mutable f
    returnStmt return
      constant 0


